test/he/test_pmtud.c

```diff
@@ -0,0 +1,269 @@
+/**
```

Comment:
Should we have a test case in this file to cover a very delayed ack coming back?  For example, I don't think `pmtud_probe_pending_id` is reset when we transition to `HE_PMTUD_STATE_SEARCH_COMPLETE`.  The ack handler's switch statement is expected to just fall through to doing nothing.
Can we confirm with a test case that it does nothing now and in the future, instead of accidentally updating some of the PMTU values?

Reply:
Good point. I think the proper fix is just reset the pending id when the `he_internal_pmtud_handle_probe_timeout`. Then no need to write complex test for the "delayed ack" scenario any more.

Reply:
I think there's still value in testing behavior one level above even if we reset the pending id.  
Just testing the internal implementation detail of "reset the pending id" is not as strong of a protection against future regressions. In the future, somebody might just decide to remove / refactor the "reset pending id logic" and its matching test case without understanding the behavior it influences.
It makes for good documentation of excepted behavior of code!

Reply:
Okay. I've ticketed it and will add later.